backend/app/controllers/api/v1/courses_controller.rb

```diff
@@ -16,6 +21,18 @@ def create
 
       private
 
+      def search_courses(courses)
+        courses = courses.where('name LIKE ?', "%#{params[:search]}%") if params[:search].present?
+        courses.map do |course|
+          {
+            name: course.name,
+            user: {
+              first_name: course.user.first_name
+            }
+          }
+        end
```

Comment:
This will lead to N+1 errors. Do you want to fix it by yourself or need some help?